Automattic/woocommerce-payments
# When COT enabled, use COT query for customer currencies query

Fixes #4148 

#### Changes proposed in this Pull Request

The code in `get_all_customer_currencies` uses DB query on `post_meta` tables to fetch all customer currencies used in orders. This function is used on the Analytics > Orders page. This will fail when COT is enabled and sync is off, as the currency data will be available only on the `orders` table. This code change fixes this so that when COT is enabled, we use an alternate query on the order tables.


#### Testing instructions

1. Ensure COT is enabled, WC Orders is set as the default table, sync is off
2. Add a new currency from **WooCommerce > Settings > Multicurrency** and place an order with that currency
3. Clear the  DB cache for `wcpay_multi_currency_customer_currencies` by deleting the entry with key `wcpay_multi_currency_customer_currencies` from options table. 
4. On **Analytics > Orders**, add the Advanced Filter "**Customer currency**". Check the currencies in the dropdown.
![image](https://user-images.githubusercontent.com/32092402/187491925-e1b39502-4600-498e-ba73-298d609fa259.png)
5. On `develop`, the new currency is not displayed in the dropdown when `orders` is the default table but will be displayed if `posts` is the default table or if sync is on.
6. On this branch, the currencies in the dropdown will be the same whether the posts or orders table is set as the authoritative table

-------------------


**Post merge**
